cms/templates/partials/card_details_top.html

```diff
@@ -19,17 +19,12 @@ <h3 class="title">
     </a>
   </h3>
   <ul>
-    {% if courseware_page.is_external_page and courseware_page.price %}
-    <li>
-      <strong>Price:</strong>
-      ${{ courseware_page.price|floatformat:"0"|intcomma }}
-    </li>
-    {% elif courseware_page.product.current_price %}
+    {% if courseware_page.product.current_price %}
```

Comment:
`CoursePage` and `ExternalCoursePage` return the `course` object; we do not have a `current_price` attribute/property in the Course Model. The `current_price` only exists for the `program pages` as the product property returns the `program` object and it has the current_price property. 

We can do:
```suggestion
    {% if courseware_page.is_internal_or_external_program_page and courseware_page.product.current_price %}
```

Reply:
I agree with utilizing the existing approach. It should be clear, as I initially thought there would be a property on the courseware pages. However, I had to investigate further to discover that the first check only applies to the program pages.

I know this is not a part of your change originally but it will improve the readability IMO.